src/compiler/interpreter.jl

```diff
@@ -55,7 +74,8 @@ function EnzymeInterpreter(
     reverse_rules::Bool,
     deferred_lower::Bool = true,
     broadcast_rewrite::Bool = true,
-    handler = nothing
+    handler = nothing,
+    local_cache = get_or_create_inference_cache(world, forward_rules, reverse_rules, deferred_lower, broadcast_rewrite, handler)
```

Comment:
If I understand correctly, the local_cache ought to be ephermal and can't be shared across multiple abstract interpreter instances.

cc: @aviatesk

Reply:
Yeah but in this case we actually are effectively caching the absint itself (since all data about it is in the key)

Reply:
All caches that ought to survive the absint go into the global cache?